### PR TITLE
[signing] dockerhub signing properly named artifacts.

### PR DESCRIPTION
**Pull request by rexhoffman**
## Motivation

For a semblance of security, let's sign our dockerhub images.   This also cleans up locations.   There's a bit of open question about validator/validator_dynamic - not sure which we want.

### Have you read the [Contributing Guidelines on pull requests]

Yes

## Test Plan

CI, is working here: https://app.circleci.com/pipelines/github/libra/libra/25188/workflows/41c54616-5d9a-4e68-ba88-7b75fdfe98da/jobs/162713

## Related PRs

None.

### CODE REVIEW

**Comment by mgorven** [comment]
Are we comfortable with the Dockerhub signing keys floating around on CircleCI?

**Comment by rexhoffman** [comment]
Tested with my testsec user as a member of libra org, with write permissions on libra, but not as an admin and not in the docker Contexts signing group, and we got this error:

Screen Shot 2020-07-29 at 2 37 36 PM

It seems contexts do what they claim.

**Comment by rexhoffman** [comment]
That means only Admins of the github libra org, or members of this group will be able to access docker creds in circle ci's actions.

This leaves the "auto" branch as pushed to via bors as the only area of risk.   Someone could  [at]bors-libra r+ and commit their own code and bors (acting on it's own) would sign the commit.

I'm pretty happy with this, and new bors will prevent even that edge case.   @mgorven a little more comfortable with this?

**Comment by rexhoffman** [comment]
Final test I needed to run passed.   Docker images signed, and sent to proper docker hub locations.

https://app.circleci.com/pipelines/github/libra/libra/25673/workflows/0d148392-ed9b-4710-9125-8e2ff0b4b2d9/jobs/165593

**Comment by rexhoffman** [comment]
@mgorven @sausagee any remaining concerns?

**Comment by mgorven** [comment]
This certainly looks better and I think this is fine for now.

File: .circleci/config.yml [hunk 1]
```diff
@@ -594,16 +608,15 @@ jobs:
...
             if [[ $IS_RELEASE  == true ]] || [[ $IS_TEST == true ]] || [[ $IS_PRE_RELEASE == true ]] ; then
-              echo "$DOCKERHUB_PASSWORD" | docker login -u "$DOCKERHUB_USERNAME" --password-stdin
               docker/update_or_build.sh -u -p -b ${BRANCH} -n client
               docker/update_or_build.sh -u -p -b ${BRANCH} -n init
               docker/update_or_build.sh -u -p -b ${BRANCH} -n mint
-              docker/update_or_build.sh -u -p -b ${BRANCH} -n safety-rules
               docker/update_or_build.sh -u -p -b ${BRANCH} -n tools
-              docker/update_or_build.sh -u -p -b ${BRANCH} -n validator-dynamic
+              docker/update_or_build.sh -u -p -b ${BRANCH} -n validator
+              docker/update_or_build.sh -u -p -b ${BRANCH} -n validator-tcb
```
Comment (sausagee) [inline .circleci/config.yml:616]: Why not just use safety-rules so that we don't have to create a mapping in the build script below?

File: docker/update_or_build.sh [hunk 2]
```diff
@@ -44,18 +44,38 @@ GIT_REV=$(git rev-parse --short=8 HEAD)
...
-#Convert dashes to underscores to get tag names, except for validator, which for some reason is "e2e"
-tag_name=`echo $NAME | sed 's/-/_/g'`
-if [ $NAME == "validator" ]; then
-  tag_name="e2e";
+#Compute the tag name generated by the docker files.
+LOCAL_TAG=`echo $INPUT_NAME | sed 's/-/_/g'`
+LOCAL_TAG=libra_${LOCAL_TAG}
+
+# BEGIN DEALING WITH INCONSISTENT NAMING
+if [ $LOCAL_TAG == "libra_validator_tcb" ]; then
```
Comment (sausagee) [inline docker/update_or_build.sh:56]: We should open an issue to clean up the tagging scheme with the final one we will go with. cc @mgorven

**Comment by mimoo** [comment]
I'm a bit late to the party, but have you documented on this works anywhere? And is signature verification included in the deployment/update process?
